src/libstore/filetransfer.cc

```diff
@@ -484,8 +484,9 @@ struct curlFileTransfer : public FileTransfer
                         code == CURLE_OK ? "" : fmt(" (curl error: %s)", curl_easy_strerror(code)))
                     : FileTransferError(err,
                         std::move(response),
-                        "unable to %s '%s': %s (%d)",
-                        request.verb(), request.uri, curl_easy_strerror(code), code);
+                        "unable to %s '%s': %s (%d)%s",
+                        request.verb(), request.uri, curl_easy_strerror(code), code,
+                        code == CURLE_SSL_CONNECT_ERROR ? "\nRecommend checking your ssl-cert-file settings,\ncreating '/etc/ssl/certs/ca-certificates.crt', or\ninstalling the 'cacert' package into '/nix/var/nix/profiles/default'." : "");
```

Comment:
Would this be also shown if someone has flaky internet and the handshake fails half way? (currently at a hotel with overloaded wifi)

Reply:
Fair point. Updated this to expose CURLOPT_ERRORBUFFER instead as it will likely be helpful in more cases.